### PR TITLE
Add a treasury growth test

### PR DESCRIPTION
**Pull request by carbolymer**
# Description

Tests that testnet genesis creates treasury which is growing in time.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs [hunk 1]
```diff
@@ -72,5 +83,69 @@ prop_foldBlocks = H.integrationRetryWorkspace 2 "foldblocks" $ \tempAbsBasePath'
   -- tests that `foldBlocks` receives ledger state; once that happens,
   -- handler is called, which then writes to the `lock` and allows the
   -- test to finish.
-  _ <- H.evalIO $ IO.readMVar lock
+  _ <- H.evalIO $ H.timeout 30_000_000 $ IO.readMVar lock
   H.assert True
+
+prop_foldBlocks_check_if_treasury_is_growing :: H.Property
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs:89]: why is this called `prop_foldBlocks_check_if_treasury_is_growing` if we're using `foldEpochState`? This should live in its own module.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs [hunk 2]
```diff
@@ -72,5 +83,69 @@ prop_foldBlocks = H.integrationRetryWorkspace 2 "foldblocks" $ \tempAbsBasePath'
...
+
+    -- | Check if the last element > first element
+    checkHasIncreased :: (Ord a) => [a] -> Bool
+    checkHasIncreased = \case
+      [] -> False
+      x1:xs -> case reverse xs of
+                [] -> False
+                xn:_ -> xn > x1
+
+    checkNonDecreasing :: (Ord a) => [a] -> Bool
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs:146]: Why not have this check in the handler? You would only need to check two consecutive epochs and you can do this by taking the length of the state `Map EpochNo Integer` and comparing the values.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs:146]: This seems to be more clear than combining the epoch treasury values and check in the same place.

It also checks that nothing weird is not going on - that treasury is suddenly not decreasing afterwards.

Moreover, that's a useful debug information to have how the treasury value was changing in the test period.

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs:89]: moved

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/FoldBlocks.hs]: Well, technically a single element set is non-decreasing. A single element set is ruled out by `checkHasIncreased`, which returns `False` for `[_]`.

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 3]
```diff
@@ -161,10 +159,16 @@ createSPOGenesisAndFiles (NumPools numPoolNodes) era shelleyGenesis (TmpAbsolute
...
 
-  H.renameFile (tempAbsPath' </> "byron-gen-command/genesis.json") (genesisByronDir </> "genesis.json")
-  -- TODO: create-testnet-data outputs the new shelley genesis do genesis.json
+  H.renameFile (tempAbsPath' </> "byron-gen-command" </> "genesis.json") (genesisByronDir </> "genesis.json")
+  -- TODO: create-testnet-data outputs the new shelley genesis to genesis.json
   H.renameFile (tempAbsPath' </> "genesis.json") (genesisShelleyDir </> "genesis.shelley.json")
 
+  -- TODO: move this to create-testnet-data
+  -- For some reason when setting "--total-supply 10E16" in create-testnet-data, we're getting negative
+  -- treasury
```
[the remark this answers is not in the document]
Comment (smelc) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:170]: Tracked here: https://github.com/IntersectMBO/cardano-cli/issues/631#issuecomment-1978254866

**Pull request merged**